tensorzero-core/src/client/mod.rs

```diff
@@ -804,64 +837,81 @@ impl Client {
         }
     }
 
+    pub async fn http_inference(
+        &self,
+        params: ClientInferenceParams,
+    ) -> Result<HttpInferenceResponse, TensorZeroError> {
+        let ClientMode::HTTPGateway(client) = &*self.mode else {
+            return Err(TensorZeroError::Other {
+                source: Error::new(ErrorDetails::InternalError {
+                    message: "HTTP inference can only be used in HTTPGateway mode".to_string(),
+                })
+                .into(),
+            });
+        };
+        let url = client
+            .base_url
+            .join("inference")
+            .map_err(|e| TensorZeroError::Other {
+                source: Error::new(ErrorDetails::InvalidBaseUrl {
+                    message: format!("Failed to join base URL with /inference endpoint: {e}"),
+                })
+                .into(),
+            })?;
+        let body = serde_json::to_string(&params).map_err(|e| TensorZeroError::Other {
+            source: Error::new(ErrorDetails::Serialization {
+                message: format!(
+                    "Failed to serialize inference params: {}",
+                    DisplayOrDebug {
+                        val: e,
+                        debug: self.verbose_errors,
+                    }
+                ),
+            })
+            .into(),
+        })?;
+        #[cfg(feature = "e2e_tests")]
+        {
+            *self.last_body.lock().await = Some(body.clone());
+        }
```

Comment:
can you add a comment about how this is used (for easy reference)?

Reply:
This was a vestige of the version-detection code - I'll remove it